nntrainer/layers/layer_node.cpp

```diff
@@ -159,6 +172,32 @@ bool LayerNode::setProperty(const std::string &key, const std::string &value) {
 
   PropertyType type = static_cast<PropertyType>(parseLayerProperty(key));
   switch (type) {
+  case PropertyType::input_shape: {
+    if (getNumInputs() > 1) {
+      throw std::invalid_argument("input_shape keyword is only for one input");
+    }
+    if (getNumInputs() == 0)
+      input_dim.resize(1);
+
+    TensorDim &in_dim = input_dim[0];
+    if (!value.empty()) {
+      unsigned int cache_batch_size = 1;
+      /** cache original value of batch size */
+      if (in_dim.batch()) {
+        cache_batch_size = in_dim.batch();
+        in_dim.batch(1);
+      }
+      int status = in_dim.setTensorDim(value.c_str());
+      if (in_dim.batch() > 1) {
+        ml_logw("Batch size set with input dimension %d is ignored."
+                "Set batchsize property for the model to update batchsize.",
+                in_dim.batch());
+      }
+      /** set back to cache value of dimension */
+      in_dim.batch(cache_batch_size);
```

Comment:
I guess it can be done later PR. If in_dim is a temporal variable then we could set it into context.

Reply:
yes, I will do this in the next PR.